src/opm/output/eclipse/WriteRFT.cpp

```diff
@@ -256,26 +344,621 @@ namespace {
+                                   ConnPos                        connPos,
+                                   const ::Opm::data::Connection& xcon);
+
+        void assignNextNeighbourID(const int id);
+
+    private:
+        PLTFlowRate flow_{};
+
+        std::vector<int> neighbour_id_{};
+
+        std::vector<float> depth_{};
+        std::vector<float> pressure_{};
+        std::vector<float> trans_{};
+        std::vector<float> kh_{};
+
+        void assignNextNeighbourID(ConnPos                       connPos,
+                                   const ::Opm::WellConnections& wellConns);
+    };
+
+    PLTRecord::PLTRecord(const std::size_t nconn)
+        : flow_{nconn}
+    {
+        if (nconn == std::size_t{0}) {
+            return;
+        }
+
+        this->neighbour_id_.reserve(nconn);
+        this->depth_.reserve(nconn);
+        this->pressure_.reserve(nconn);
+        this->trans_.reserve(nconn);
+        this->kh_.reserve(nconn);
+    }
+
+    void PLTRecord::collectRecordData(const ::Opm::UnitSystem&  usys,
+                                      const ::Opm::EclipseGrid& grid,
+                                      const ::Opm::Well&        well,
+                                      const ::Opm::data::Well&  wellSol)
+    {
+        this->prepareConnections(well);
+
+        const auto& xcon = wellSol.connections;
+        connectionLoop(well.getConnections(), grid,
+            [this, &usys, &well, &xcon](ConnPos connPos)
+        {
+            const auto xconPos =
+                findConnResults(connPos->global_index(), xcon);
+
+            if (! xconPos.has_value()) {
+                return;
+            }
+
+            this->addConnection(usys, well, connPos, *xconPos.value());
+        });
+    }
+
+    void PLTRecord::write(::Opm::EclIO::OutputStream::RFT& rftFile) const
+    {
+        rftFile.write("CONDEPTH", this->depth_);
+        rftFile.write("CONPRES" , this->pressure_);
+
+        rftFile.write("CONORAT", this->flow_.oil());
+        rftFile.write("CONWRAT", this->flow_.water());
+        rftFile.write("CONGRAT", this->flow_.gas());
+
+        rftFile.write("CONFAC", this->trans_);
+        rftFile.write("CONKH" , this->kh_);
+        rftFile.write("CONNXT", this->neighbour_id_);
+    }
+
+    void PLTRecord::prepareConnections([[maybe_unused]] const ::Opm::Well& well)
+    {}
+
+    void PLTRecord::addConnection(const ::Opm::UnitSystem&       usys,
+                                  const ::Opm::Well&             well,
+                                  ConnPos                        connPos,
+                                  const ::Opm::data::Connection& xcon)
+    {
+        using M = ::Opm::UnitSystem::measure;
+        auto cvrt = [&usys](const M meas, const double x) -> float
+        {
+            return usys.from_si(meas, x);
+        };
+
+        // Allocate neighbour ID element
+        this->neighbour_id_.push_back(0);
```

Comment:
> Why not simply use `push_back()` in `assignNextNeighbourId(int)`?

Because that would lead to calling `push_back()` twice for multi-segmented wells.  Those are, in the context of creating an RFT file PLT record, a pure extension of the regular/standard wells so I elected to implement the type through public inheritance from the PLT record for regular wells.  The base type will call `assignNextNeighbour(int)` for the regular part, and then the MSW type will call the function again to handle the MSW part.